sympy/sets/ordinals.py

```diff
@@ -32,24 +30,41 @@ def _compare_term(self, other, op):
         else:
             return op(self.exp, other.exp)
 
+    def __eq__(self, other):
+        if not isinstance(other, OmegaPower):
+            try:
+                other = Ordinal.convert(other).args[0]
```

Comment:
Perhaps this could be `OmegaPower(0, other)` if proper conversions are made in `__new__`. (`0` should then be converted to ordinal 0, which is `Ordinal()` with no arguments. That could probably also be a singleton `class Ord0` created once for all.)